src/modules/simulation/gz_bridge/CMakeLists.txt

```diff
@@ -82,21 +82,74 @@ if(gz-transport_FOUND)
 		${PX4_SOURCE_DIR}/Tools/simulation/gz/worlds/*.sdf
 	)
 
+	# find corresponding airframes
+	file(GLOB gz_airframes
+	     RELATIVE ${PX4_SOURCE_DIR}/ROMFS/px4fmu_common/init.d-posix/airframes
+	     ${PX4_SOURCE_DIR}/ROMFS/px4fmu_common/init.d-posix/airframes/*_gz_*
+	)
+
+	# remove any .post files
+	foreach(gz_airframe IN LISTS gz_airframes)
+		if(gz_airframe MATCHES ".post")
+			list(REMOVE_ITEM gz_airframes ${gz_airframe})
+		endif()
+	endforeach()
+	list(REMOVE_DUPLICATES gz_airframes)
+
+	foreach(gz_airframe IN LISTS gz_airframes)
+		set(model_only)
+		string(REGEX REPLACE ".*_gz_" "" model_only ${gz_airframe})
+
+		if(EXISTS "${PX4_SOURCE_DIR}/Tools/simulation/gz/models/${model_only}")
+
+			if((EXISTS "${PX4_SOURCE_DIR}/Tools/simulation/gz/models/${model_only}/model.sdf"))
+				#message(STATUS "Ignition SDF file found for ${model_only}")
+			else()
+				message(WARNING "Ignition no SDF file found for ${model_only}")
+			endif()
+
+		else()
+			message(WARNING "model directory ${PX4_SOURCE_DIR}/Tools/simulation/gz/models/${model_only} not found")
+		endif()
+	endforeach()
+
 	foreach(model ${gz_models})
+
+		# match model to airframe
+		set(airframe_model_only)
+		set(airframe_sys_autostart)
+		set(gz_airframe_found)
+		foreach(gz_airframe IN LISTS gz_airframes)
+
+			string(REGEX REPLACE ".*_gz_" "" airframe_model_only ${gz_airframe})
+			string(REGEX REPLACE "_gz_.*" "" airframe_sys_autostart ${gz_airframe})
+
+			if(model STREQUAL ${airframe_model_only})
+				set(gz_airframe_found ${gz_airframe})
+				break()
+			endif()
+		endforeach()
+
+		if(gz_airframe_found)
+			#message(STATUS "gz model: ${model} (${airframe_model_only}), airframe: ${gz_airframe_found}, SYS_AUTOSTART: ${airframe_sys_autostart}")
+		else()
+			message(WARNING "gz missing model: ${model} (${airframe_model_only}), airframe: ${gz_airframe_found}, SYS_AUTOSTART: ${airframe_sys_autostart}")
+		endif()
+
 		foreach(world ${gz_worlds})
 
 			get_filename_component("world_name" ${world} NAME_WE)
 
-			if(world_name MATCHES "default")
+			if(world_name STREQUAL "default")
 				add_custom_target(gz_${model}
-					COMMAND ${CMAKE_COMMAND} -E env PX4_SIM_MODEL=${model} $<TARGET_FILE:px4>
+					COMMAND ${CMAKE_COMMAND} -E env PX4_SIM_MODEL=gz_${model} $<TARGET_FILE:px4>
 					WORKING_DIRECTORY ${SITL_WORKING_DIR}
 					USES_TERMINAL
 					DEPENDS px4
 				)
 			else()
 				add_custom_target(gz_${model}_${world_name}
-					COMMAND ${CMAKE_COMMAND} -E env PX4_GZ_MODEL=${model} PX4_GZ_WORLD=${world_name} $<TARGET_FILE:px4>
+					COMMAND ${CMAKE_COMMAND} -E env PX4_SIM_MODEL=gz_${model} PX4_SIM_WORLD=${world_name} $<TARGET_FILE:px4>
```

Comment:
@dagar , to define the world file `PX4_GZ_WORLD` is need, not `PX4_SIM_WORLD`.

Reply:
Hmm, this is getting kind of convoluted, maybe we need to back up a bit and review/document it centrally before it gets any worse.

Reply:
Well, at least for gazebo ignition is it quite documented. Do you mean adding docs here or in https://docs.px4.io/?

And what is the role of `PX4_SIM_WORLD`? It is defined here for classic gazebo https://github.com/PX4/PX4-Autopilot/blob/main/Tools/simulation/gazebo/sitl_run.sh#L71 and here for jsbsim https://github.com/PX4/PX4-Autopilot/blob/main/Tools/simulation/jsbsim/sitl_run.sh#L41 but it does not appear to be used.

Reply:
> And what is the role of PX4_SIM_WORLD

Yes that's what I'm getting at.